svc.go

```diff
@@ -212,6 +214,14 @@ func (as *autoNATService) doDial(pi peer.AddrInfo) *pb.Message_DialResponse {
 
 // Enable the autoNAT service temporarily until the associated context is canceled.
 func (as *autoNATService) Enable(ctx context.Context) {
+	alreadyRunning := atomic.SwapUint32(&as.running, 1)
+	if alreadyRunning > 0 {
+		return
+	}
+	defer func() {
+		atomic.StoreUint32(&as.running, 0)
```

Comment:
If we're actually relying on this, this is racy. If the service sees the cancel after we try to enable, we'll end up shutting it down.

Reply:
I believe this is never actually 'racy' - in the case when the service is force-enabled, a long-lived instance will run with the main context in the constructor.

In the case when it is running when public, it will not be created until after the startup delay has passed, and the only one instance will be attempted at a time, managed by the main autonat event loop.

This 'running' flag prevents both from happening at once.

Would it be clearer to remove the atomic?

Reply:
If, for some reason, we switch from reachable to unreachable and back to reachable _really_ fast, we'll fail to start the service. I think the _right_ way to do this is to have a start/stop function that waits.

Does this sound right?

Reply:
makes sense. updated to that structure.